src/Entity/OgRole.php

```diff
@@ -6,177 +6,206 @@
  */
 namespace Drupal\og\Entity;
 
-use Drupal\Core\Entity\ContentEntityBase;
-use Drupal\Core\Entity\EntityTypeInterface;
-use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Config\Entity\ConfigEntityBase;
 
 /**
- * todo: Find a way to attach the roles to group:
- *  - UUID?
- *  - Service?
- *
- * Convert into
- *
- *
- * @ContentEntityType(
+ * @ConfigEntityType(
  *   id = "og_role",
  *   label = @Translation("OG role"),
- *   module = "og",
- *   base_table = "og_role",
+ *   static_cache = TRUE,
  *   entity_keys = {
- *     "id" = "rid",
- *     "label" = "name"
+ *     "id" = "id",
+ *     "label" = "label",
+ *     "group_type" = "groupType",
+ *     "group_bundle" = "groupBundle",
+ *     "uid" = "uid",
+ *     "permissions" = "permissions"
  *   },
+ *   config_export = {
+ *     "id",
+ *     "label",
+ *     "group_type",
+ *     "group_bundle",
+ *     "uid",
+ *     "permissions"
+ *   }
  * )
  */
-class OgRole extends ContentEntityBase {
+class OgRole extends ConfigEntityBase {
```

Comment:
Since we are modeling this after `\Drupal\user\Entity\Role` we should follow it as much as possible, and add also an `OgRoleInterface`. I'm on it :)